R/single_value.R

```diff
@@ -0,0 +1,29 @@
+#' Ensure that a vector has only a single value throughout.
+#'
+#' Missing values are replaced with the single value, and if all values are
+#' missing, the first value in \code{missing} is used throughout.
+#'
+#' @param x The vector which should have a single value
+#' @param missing The vector of values to consider missing in \code{x}
+#' @param warn_if_all_missing Generate a warning if all values are missing?
+#' @param info If more than one value is found, append this to the warning or
+#'   error to assist with determining the location of the issue.
+#' @return \code{x} as the scalar single value found throughout (or an error if
+#'   more than one value is found).
+#' @examples
+#' single_value(c(NA, 1))
+#' @export
+single_value <- function(x, missing=NA, warn_if_all_missing=FALSE, info=NULL) {
```

Comment:
I don't see `warn_if_all_missing` used in the function, e.g., `single_value(c(""), missing = c(NA, ""), warn_if_all_missing = TRUE)` does not warn.  But I think it's a useful feature to implement.

Reply:
Whoops!  Fixed and tested.